upstairs/src/downstairs.rs

```diff
@@ -71,7 +71,9 @@ pub(crate) struct Downstairs {
     /// Indicates whether we are eligible for replay
     ///
     /// We are only eligible for replay if all jobs since the last flush are
-    /// buffered (i.e. none have been retired by a `Barrier` operation).
+    /// buffered (i.e. none have been retired by a `Barrier` operation), or we
```

Comment:
I'd suggest leaving this comment as-is; we don't actually change the value of `self.can_replay` when live-repair starts, so the new comment is misleading.  You could move the new part of the comment below (before calling `reinitialize`), e.g. something like
```rust
        // Restart the IO task for that specific client, transitioning to a new
        // state.  We can replay as long as all jobs since the last flush are
        // buffered (indicated by `self.can_replay`), and there's no live-repair
        // in progress.
        let can_replay = self.can_replay && !self.live_repair_in_progress();
        self.clients[client_id].reinitialize(up_state, can_replay);
```